testing-modules/mockito-2/src/main/java/com/baeldung/mockito/objectmapper/FlowerJsonStringValidator.java

```diff
@@ -0,0 +1,21 @@
+package com.baeldung.mockito.objectmapper;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class FlowerJsonStringValidator {
+    private ObjectMapper objectMapper;
+
+    public FlowerJsonStringValidator(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public boolean flowerHasPetals(String jsonFlowerAsString) throws JsonProcessingException {
+        Flower flower = readFlower(jsonFlowerAsString);
+        return flower.getPetals() > 0;
+    }
+
+    private Flower readFlower(String flowerAsJson) throws JsonProcessingException {
+        return objectMapper.readValue(flowerAsJson, Flower.class);
```

Comment:
Since we only use this once and it's a one-liner, let's just move it to the flowerHasPetals method to simplify the call stack.